src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -73,26 +73,41 @@ public Secret getPrivateKey() {
     }
 
     @SuppressWarnings("deprecation") // preview features are required for GitHub app integration, GitHub api adds deprecated to all preview methods
-    static String generateAppInstallationToken(String appId, String appPrivateKey, String apiUrl) {
-        try {
-            String jwtToken = createJWT(appId, appPrivateKey);
-            GitHub gitHubApp = new GitHubBuilder().withEndpoint(apiUrl).withJwtToken(jwtToken).build();
+    @CheckForNull
+    GHAppInstallation getAppInstallation() {
+        String jwtToken = createJWT(appID, privateKey.getPlainText());
+        GitHubBuilder ghb = new GitHubBuilder();
+        if (Util.fixEmpty(apiUri) != null) {
+            ghb = ghb.withEndpoint(apiUri);
+        }
+        GitHub gitHubApp = ghb.withJwtToken(jwtToken).build();
+
+        GHApp app = gitHubApp.getApp();
 
-            GHApp app = gitHubApp.getApp();
+        List<GHAppInstallation> appInstallations = app.listInstallations().asList();
+        if (appInstallations.isEmpty()) {
+            return null;
+        } else {
+            // TODO make sensitive to contextual organization
```

Comment:
see https://github.com/jenkinsci/github-branch-source-plugin/pull/269#discussion_r395887018

Reply:
Currently working on this, almost got it done, api calls are working but for some reason can't get checkout the repo.